packages/manager/modules/vps/src/migration/vps-migration.service.js

```diff
@@ -0,0 +1,96 @@
+import compact from 'lodash/compact';
+import find from 'lodash/find';
+import map from 'lodash/map';
+import set from 'lodash/set';
+
+import { EXCLUDED_VERSIONS } from './vps-migration.constants';
+
+export default class VpsMigrationService {
+  /* @ngInject */
+  constructor($q, $http, $translate, $timeout) {
+    this.$q = $q;
+    this.$http = $http;
+    this.$translate = $translate;
+    this.$timeout = $timeout;
+  }
+
+  getVpsList() {
+    return this.$http
+      .get(`/vps`)
+      .then(({ data }) =>
+        this.$q.all(map(data, (vps) => this.getVpsDetails(vps))),
+      )
+      .then((list) =>
+        compact(
+          map(list, ({ data }) => {
+            if (EXCLUDED_VERSIONS.includes(data.model.version)) {
+              return data;
+            }
+            return null;
+          }),
+        ),
+      );
+  }
+
+  getVpsDetails(serviceName) {
+    return this.$http.get(`/vps/${serviceName}`);
+  }
+
+  getMigrationDetails(serviceName, catalog) {
+    return this.$http
+      .get(`/vps/${serviceName}/migration2014`)
+      .then((details) =>
+        VpsMigrationService.populateOptionsPrice(details.data, catalog),
+      );
+  }
+
+  scheduleMigration(serviceName, date) {
+    return this.$http.post(`/vps/${serviceName}/migration2014`, { date });
+  }
+
+  static getPriceWithTax(price) {
+    return (price.price + price.tax) / 100000000;
+  }
+
+  static populateOptionsPrice(vpsMigration, catalog) {
+    let totalOptionsPrice = 0;
+    map(vpsMigration.options, (option) => {
+      const prices = find(catalog.addons, {
+        planCode: option.vps2020code,
+      });
+      const price = find(prices.pricings, { mode: 'default', interval: 1 });
+      set(option, 'price', VpsMigrationService.getPriceWithTax(price));
+      totalOptionsPrice += option.price;
+      return option;
+    });
+    set(vpsMigration, 'totalOptionsPrice', totalOptionsPrice);
+    return vpsMigration;
+  }
+
+  static getMigrationPlan(catalog, name) {
+    const product = find(catalog.products, { name });
+    const plan = find(catalog.plans, { planCode: name });
+    const price = find(plan.pricings, { mode: 'default', interval: 1 });
+    return {
+      description: product.description,
+      price: VpsMigrationService.getPriceWithTax(price),
+    };
+  }
+
+  getVpsModelDescription(model) {
```

Comment:
Nitpicking: Could be exposed as a getter from a VPS instance and declared in a `Vps.class.js`

Reply:
Actually this function is for 2 different type of objects, one for VPS Detail and one for migration Plan. Both are different and wont fit in a single class

Reply:
I think it possible, based on the object we can return the description.